src/main/java/org/apache/solr/benchmarks/indexing/UploadDocs.java

```diff
@@ -99,9 +102,12 @@ public void writeTo(OutputStream outstream) throws IOException {
 
         HttpResponse rsp = client.execute(httpPost);
         int statusCode = rsp.getStatusLine().getStatusCode();
-        if (statusCode != 200) {
+        if (!HttpStatus.isSuccess(statusCode)) {
```

Comment:
Nice! It would be nice here to have retries for connection-related error codes, but maybe that's a different PR 😅